File: Day 9/javascript/auction.js

let keep_going = true;
let bidders = {};

console.log("Welcome to the secret auction program.");

function highest_bid() {
    let count = 0;
    let high_key = "";

    for (const [key, value] of Object.entries(bidders)) {
        if (value > count) {
            count = value;
            high_key = key;
        };
    };
    console.log("The winner is " + high_key + " with a bid of $" + count + ".");
}

while (keep_going) {

    let name_1 = prompt("What is your name? ");
    let bid = Number(prompt("What is your bid? $"));

    bidders[name_1] = bid;

    let keep_pushing = prompt("Are there any other bidders? Type 'yes' or 'no'.");

    if (keep_pushing == 'no') {
        keep_going = false;
        highest_bid();
    }; 
};

File: Day 9/python/auction.py
# form replit import art   # this is for replit to clear the terminal

import art

keep_going = True
bidders = {}

print(art.logo)
print("Welcome to the secret auction program.")

def highest_bid():
    count = 0
    high_key = ""
    for key in bidders:
        if bidders[key] > count:
            count = bidders[key]
            high_key = key
            
    print(f"The winner is {high_key} with a bid of ${count}.")

while keep_going:

    name = input("What is your name?: ")
    bid = int(input("What is your bid?: $"))

    bidders[name] = bid

    keep_pushing = input("Are there any other bidders? Type 'yes' or 'no'.\n")

    if keep_pushing == 'no':
        keep_going = False
        highest_bid()
    #elif keep_pushing == 'yes': # this is where we would clear if using replit.
    #    clear()

